### ISSUE TITLE
Unexpected response in a simple search

### ISSUE THREAD

**Issue by brunoocasali** [meilisearch/meilisearch#2722]
**Describe the bug**
Working to add support in the Meilisearch Ruby SDK to v0.29 I found the following situation:

**To Reproduce**
With this database:
```ruby
{"results"=>
  [{"objectId"=>123, "title"=>"Pride and Prejudice", "genre"=>"romance"},
   {"objectId"=>456, "title"=>"Le Petit Prince", "genre"=>"adventure"},
   {"objectId"=>1, "title"=>"Alice In Wonderland", "genre"=>"adventure"},
   {"objectId"=>2, "title"=>"Le Rouge et le Noir", "genre"=>"romance"},
   {"objectId"=>1344, "title"=>"The Hobbit", "genre"=>"adventure"},
   {"objectId"=>4, "title"=>"Harry Potter and the Half-Blood Prince", "genre"=>"fantasy"},
   {"objectId"=>42, "title"=>"The Hitchhiker's Guide to the Galaxy"}],
 "offset"=>0,
 "limit"=>20,
 "total"=>7}
```

When I do a phrase search request `index.search('coco "harry"')` 

It returns nothing.

```ruby
=> {"hits"=>[], "estimatedTotalHits"=>0, "query"=>"coco \"harry\"", "limit"=>20, "offset"=>0, "processingTimeMs"=>0, "nbHits"=>0}
```

**Expected behavior**

On version 0.28 the results are:

```ruby
=> {"hits"=>[{"objectId"=>4, "title"=>"Harry Potter and the Half-Blood Prince", "genre"=>"fantasy"}], "estimatedTotalHits"=>1, "query"=>"coco \"harry\"", "limit"=>20, "offset"=>0, "processingTimeMs"=>0, "nbHits"=>1}
```

**Meilisearch version:** v0.29.0rc1, docker.

**Comment by ManyTheFish** [comment]
Hello @Kerollmops and @brunoocasali, here we are facing an undefined behavior, the `word` ranking rule removes words from the left to the right, but, because there is a phrase query, we don't remove it. Instead, we remove the next "non-phrased word".

However, as you said the behavior has changed when we only have one "non-phrased word". The previous behavior seemed to always keep the first written word.

Which one should be considered bugged? 😬

poke @gmourier

**Comment by curquiza** [comment]
Closed by #2732
